src/shared/use-data-value-set/use-data-value-set.js

```diff
@@ -45,24 +45,25 @@ function mapDataValuesToFormInitialValues(dataValues) {
     return formInitialValues
 }
 
+const select = (data) => {
+    const dataValues = mapDataValuesToFormInitialValues(data.dataValues)
+    const minMaxValues = data.minMaxValues || {}
+    return { dataValues, minMaxValues }
+}
```

Comment:
```suggestion
const select = createSelector(
    (data) => data,
    (data) => {
        const dataValues = mapDataValuesToFormInitialValues(data.dataValues)
        const minMaxValues = data.minMaxValues || {}
        return { dataValues, minMaxValues }
    }
)
```

Reply:
According to the [documentation](https://tkdodo.eu/blog/react-query-data-transformations#3-using-the-select-option), my solution should work:

> Selectors like the one above will also run on every render, because the functional identity changes (it's an inline function)
> If your transformation is expensive, you can memoize it either with useCallback, or by extracting it to a stable function reference

I moved it outside of the component, giving it a stable reference

Reply:
That hook is only being used in two places (`data-workspace.js` & `use-highlighted-field.js`), the fields don't use it anymore. So I'm wondering what's more expensive:
* Re-computing this when the data values change
* checking if the data has changed on every render

I suppose the re-computing as an operation itself is more expensive, but checking for a change happens more often, so it could be more expensive the accumulative time it take on every render. Unless this is a real performance issue, I wouldn't use reselect here. The data structure of this request seems simple to me, unlike the metadata select callback for example.

I don't mind if we profile this though, if you're worried about this!

Reply:
> I moved it outside of the component, giving it a stable reference. That means it won't run the select function on re-renders.

It's not enough, just test to print something in the selector. It's atleast run for every mount.

We are doing this for other selectors, like in use-metadata.


> checking if the data has changed on every render

It's a simple reference check, so I wouldn't worry about this at all.

Reply:
But yeah I tested it some and it doesn't seem like it is recomputed that much, but it is atleast computed 3 times in app load.

Reply:
Alright, I've wrapped it